tests/parsing.cpp

```diff
@@ -0,0 +1,234 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parsing.cpp                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: rmorel <rmorel@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/06/22 16:47:06 by rmorel            #+#    #+#             */
+/*   Updated: 2023/06/28 17:27:27 by rmorel           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "constant.h"
+#include "core/color.h"
+# include "tests.h"
+# include "Message.hpp"
+# include "error.h"
+
+int parsing_test_cmd(t_cmd_type expected_cmd, 
+		std::vector<std::string> expected_params,
+		std::map<std::string, std::string> expected_tags,
+		std::string expected_prefix,
+		Message &message)
+{
+	message.print_message();
+	DEBUG("After printing");
+	if(message.get_tags() != expected_tags)
```

Comment:
Pour tous tes tests te fais meme pas chier à faire des if puis des return, fait direct des assert genre comme ça:
```suggestion
	assert(message.get_tags() != expected_tags);
```
Genre les tests, c'est vraiment tu verifis que ton truc donne le resultat que tu veux, sinon cest tchao

Reply:
Bien vu, j'avais du mal a voir comment organiser mes tests mais c'est bien mieux comme ça !